GameServerLib/Logic/GameObjects/AttackableUnits/AI/Champion.cs

```diff
@@ -17,6 +17,8 @@ public class Champion : ObjAiBase
         public float ChampionGoldFromMinions { get; set; }
         public RuneCollection RuneList { get; set; }
         public Dictionary<short, Spell> Spells { get; private set; } = new Dictionary<short, Spell>();
+        public ChampionStats ChampStats = new ChampionStats();
```

Comment:
Make it private

Reply:
But minion class should change some of the stats, because for example when minion die some of the stats of the champion should be changed.

It's possible to make functions for each change (something like 20 functions), but I saw that Stats in AttackableUnit are also Public so I kept ChampStats public as well.

Reply:
In cases like that you add a function to the variable holder `Champion in this case` which does what you want, and call that from the minion for example. Generally it's bad practice to modify another object's variables.

Reply:
Ok, I'll write OnKillMinion function in Champion.cs to do that, and make ChampStats property with no set (public to read).

Reply:
@MythicManiac I only half-agree with you on that one, C# and properties were especially made for this. With proper synchronization you can do good stuff